### PR TITLE
Add doc directory with API.md

### PR DESCRIPTION
**Pull request by edouardparis**

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Note that all addresses are bech32-encoded *version 0* native Segwit `scriptPubKey`s.
+
+| Command                                                     | Description                                          |
+| ----------------------------------------------------------- | ---------------------------------------------------- |
+| [`help`](#help)                                             | Display all available commands                       |
+| [`stop`](#stop)                                             | Stops the minisafe daemon                            |
+| [`getinfo`](#getinfo)                                       | Display general information                          |
+| [`getaddress`](#getaddress)                                 | Get an address                                       |
+| [`listutxos`](#listutxos)                                   | Display a paginated list of utxos                    |
```
Comment (darosior) [inline doc/API.md:14]: Can we be more generalistic? Maybe a `gethistory`? Or `listtransactions`?

Ack for pagination.

File: doc/API.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+### `help`
+
+Display all available commands.
+
+#### Response
+
+| Field      | Type   | Description                                                                                                      |
+| ---------- | ------ | ---------------------------------------------------------------------------------------------------------------- |
+| `commands` | object | One entry per command, specifying the command name and parameters. Optional parameters are enclosed in brackets. |
```
Comment (darosior) [inline doc/API.md:30]: How about

```json
{
  "commands": [
    {
      "name": "getinfo",
      "parameters": [
        {
          "name": "blablbla",
          "optional": false
        }
      ]
    },
    ....
  ]
}
```

File: doc/API.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+### `getaddress`
+
+Get an address.
+
+#### Response
+
+| Field         | Type              | Description                                                 |
+| ------------- | ----------------- | ----------------------------------------------------------- |
+| `index`       | string (optional) | Get a deposit address for a specific derivation index       |
```
Comment (darosior) [inline doc/API.md:57]: How about `getnewaddress` and no (officially documented) `index` parameter?

File: doc/API.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+| Field               | Type         | Description                                |
+| ------------------- | ------------ | ------------------------------------------ |
+| `psbt`              | string       | Base64-encoded Spend transaction PSBT      |
+| `change_index`      | integer      | Index of the change output, might be null  |
+| `status`            | string       | [Spend status](#spend-status)              |
+
+### `broadcast`
+
+The `broadcast` RPC Command build and broadcast the transaction out of
+a signed psbt.
```
Comment (darosior) [inline doc/API.md:128]: How about an `updatetx` command that takes two parameters:
1. The PSBT
2. An optional boolean: "`broadcast`"

So this already supports multisigs

File: doc/API.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+### `getspendtx`
+
+The `getspendtx` RPC Command builds and returns the spend transaction given a
+set of utxos to spend.
+
+#### Request
+
+| Parameter   | Type                 | Description                        |
+| ----------- | -------------------- | ---------------------------------- |
+| `inputs`    | string array         | Utxo outpoints                     |
```
Comment (darosior) [inline doc/API.md:93]: Should be optional IMO

File: doc/API.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+| `psbt`     | [Spend transaction resources](#spend_transaction_resources) | Spend transaction informations |
+
+
+##### Spend status
+
+Please note that this status refers only to the Spend transaction, with regarding to the signatures and the broadcast status.
+
+| Value           | Description                                                                                      |
+| --------------- | ------------------------------------------------------------------------------------------------ |
+| `non_final`     | The Spend transaction is not final, we are awaiting signatures either from managers or cosigners |
```
Comment (darosior) [inline doc/API.md:110]: copypasta

Comment (edouardparis) [inline doc/API.md]: remove

**Review by darosior — COMMENTED** [review]
Cool. Thanks for starting this. Do we need a specifc call for cycling the coins?

File: doc/API.md (hunk 1 above)
Comment (edouardparis) [reply doc/API.md:14]: We need both, a listutxos that help the gui to calculate the current balance and say which coins need cycling and a gettransactions or gethistory to see what happened

Comment (edouardparis) [reply doc/API.md:14]: So pagination for gethistory/gettransactions, but not for listutxos (my bad)

File: doc/API.md (hunk 4 above)
Comment (edouardparis) [reply doc/API.md:120]: I would prefer `updatepsbt`and we will need `listpsbts` and `deletepsbt`

Comment (edouardparis) [reply doc/API.md:169]: I changed for `updatespendtx`, listspendtxs, deletespendtx

File: doc/API.md (hunk 1 above)
Comment (darosior) [reply doc/API.md:14]: We could also add filtering to `listutxos` eventually.

nit: how about naming it `listcoins`?

File: doc/API.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+| ----------------------------------------------------------- | ---------------------------------------------------- |
+| [`help`](#help)                                             | Display all available commands                       |
+| [`stop`](#stop)                                             | Stops the minisafe daemon                            |
+| [`getinfo`](#getinfo)                                       | Display general information                          |
+| [`getaddress`](#getaddress)                                 | Get an address                                       |
+| [`listutxos`](#listutxos)                                   | Display a paginated list of utxos                    |
+| [`getspendtx`](#getspendtx)                                 | Retrieve the Minisafe spend transaction to sign      |
+| [`updatespendtx`](#updatespendtx)                           | Update the Minisafe spend transaction                |
+| [`deletespendtx`](#deletespendtx)                           | Delete the Minisafe spend transaction                |
+| [`listspendtxs`](#listspendtxs)                             | Retrieve the Minisafe pending outgoing transactions  |
```
Comment (darosior) [inline doc/API.md:18]: I agree it makes sense to have the same concept of pending Spend txs as in revaultd. Would be worth discussing it with @kloaec too.

Comment (darosior) [reply doc/API.md:18]: Also, let's name it with `psbt` rather than `spendtx`?

**Comment by darosior** [comment]
I think we should take smaller steps rather than introducing the bulk of the API doc at once.

**Comment by edouardparis** [comment]
doc/API.md is more up to date on `master` now

**Pull request closed without merging**